core/app/c/[communitySlug]/stages/manage/components/panel/actionsTab/StagePanelRuleCreator.tsx

```diff
@@ -196,6 +318,26 @@ export const StagePanelRuleCreator = (props: Props) => {
 										</FormItem>
 									)}
 								/>
+
+								{/* Additional selector for watched action when using action chaining events */}
+								{isActionChainingEvent && (
+									<FormField
+										control={form.control}
+										name="watchedActionInstanceId"
+										render={({ field }) => (
+											<ActionSelector
+												fieldProps={field}
+												actionInstances={props.actionInstances}
+												label="after action..."
+												placeholder="Select action to watch"
+												key={field.value}
+												disabledActionId={selectedActionInstanceId} // Prevent self-references
```

Comment:
maybe we could do more to show the user what combinations are not allowed (cycles, max depth, already existing rules) before the user submits, but for now i think this is fine. but if we do want better client feedback (other than a toast) let me know!